### ISSUE TITLE
Release 4.4.2 - Release Candidate 1 - External Integrations modules

### ISSUE THREAD

**Issue by EduLeon12** [wazuh/wazuh#16974]
The following issue aims to perform all the required testing for the current release candidate to ensure the modules (AWS, GCP, Azure, DockerListener and Shuffle) works as expected, report the results, and open new issues for any encountered errors.

## Modules tests information

|||
|----------------------------------|------ |
| **Main release candidate issue** | #16966    |
| **Version**                      | 4.4.2 |
| **Release candidate #**          |   RC1  |
| **Tag**                          | [v4.4.2-rc1](https://github.com/wazuh/wazuh/tree/v4.4.2-rc1)|
| **Previous modules tests issue** | #16598  |

## Test report procedure

All test results must have one of the following statuses: 
|                                  |                                                        |
|-----------------|-------------------------------------------------------------------------|
| :green_circle:  | The test passed successfully. 										    |
| :yellow_circle: | The test passed but some enhancements must be applied to the module. |
| :red_circle:    | The test failed.                                                        |
| :white_circle:  | The test does not apply.                                                |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the `Fixes` section of the current release candidate main issue.

The resulting logs for the tests must be included in the status report so it can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

| **Module**      |  **Status**  |**Notes**|
|-----------------|-------------|---------|
| AWS             | 🟢    |  |
| Azure           |  🟢  |  |
| GCP             |  :white_circle:  | Skipped |
| Docker Listener |  🟢    |  |
| Shuffle |  🟢    |  |

All checks from below must be accepted in order to close this issue.
- [ ] @davidjiglesias

**Comment by EduLeon12** [comment]
# AWS Test results

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected amount of events were sent to `analysisd`.
- The expected output was present in the `{wazuh_path}/logs/ossec.log` file.

The data available in our Wazuh-dev environment for each of these supported services will be used. Additional logs may be created if they are necessary or there is insufficient data for testing.

## Test results

| Bucket/Service type            | Status                  | Issue ref. |
|--------------------------------|-------------------------|------------|
| Test using ossec.conf in a manager | -  | -          |
| Test using ossec.conf in an agent (Python 3.9)   | -  | -          |

## Test using ossec.conf in a manager

<details>
    <summary>Configuration</summary>

  <wodle name="aws-s3">
    <disabled>no</disabled>
    <interval>1h</interval>
    <run_on_start>yes</run_on_start>
    <subscriber type="security_lake">
         <sqs_name>test_sqs_name</sqs_name>
         <iam_role_arn>test_role</iam_role_arn>
         <external_id>test_external_id</external_id>
         <iam_role_duration>1300</iam_role_duration>
    </subscriber>
</wodle>


</details>

**Comment by EduLeon12** [comment]
# GCP Test results

No data must be modified or removed unless it was specified for the test.

The tests assume there is a valid authentication file located at `/var/ossec/wodles/gcloud/credentials.json`.

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected amount of events were sent to `analysisd`.
- The expected output was present in the `{wazuh_path}/logs/ossec.log` file.

## Test results

| Bucket/Service type            | Status                  | Issue ref. |
|--------------------------------|-------------------------|------------|
| Test using ossec.conf in a manager |   | -          |
| Test using ossec.conf in a agent (Python 3.6)   |   | -          |

## Environment details

To test Pub/Sub integration a script was used to publish any number of messages in the testing topic so we can ensure there always will be data available for the tests. This was required because of how Pub/Sub is designed as the messages are consumed once they are pulled.

**Comment by mhamra** [comment]
# Azure Test results
### Environment preparation

1. Clone git branch [v4.4.2-rc1](https://documentation-dev.wazuh.com/v4.4.2-rc1) from wazuh repo into local folder
2. Environment creation using docker from repo wazuh-tools/framework/environment/dev
    - Configure the .env file and recreate all environment's images with "docker compose up --build -d"
```
MANAGER_BRANCH=v4.4.2-rc1
MANAGER_VERSION=4.4.2
AGENT_BRANCH=v4.4.2-rc1
AGENT_VERSION=4.4.2
WAZUH_PYTHON_VERSION=3.9.16
WAZUH_LOCAL_PATH=/home/eduardoleon/git/wazuh
```
Results: the master node shows that **version 4.4.2** it's being used
```
2023-05-03 09:11:18 2023/05/03 12:11:18 wazuh-db[39] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2023-05-03 09:11:18 2023/05/03 12:11:18 wazuh-db[39] main.c:119 at main(): DEBUG: Wazuh home directory: /var/ossec
2023-05-03 09:11:19 2023/05/03 12:11:19 wazuh-remoted[44] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2023-05-03 09:11:19 2023/05/03 12:11:19 wazuh-remoted[44] main.c:130 at main(): DEBUG: Wazuh home directory: /var/ossec
2023-05-03 09:11:19 2023/05/03 12:11:19 wazuh-remoted[44] main.c:148 at main(): DEBUG: This is not a worker
2023-05-03 09:11:21 2023/05/03 12:11:21 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
2023-05-03 09:11:21 2023/05/03 12:11:21 wazuh-authd[110] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2023-05-03 09:11:21 2023/05/03 12:11:21 wazuh-authd[110] config.c:23 at authd_read_config(): DEBUG: Reading configuration 'etc/ossec.conf'
2023-05-03 09:11:21 2023/05/03 12:11:21 wazuh-authd[110] main-server.c:343 at main(): DEBUG: Wazuh home directory: /var/ossec
2023-05-03 09:11:21 2023/05/03 12:11:21 wazuh-db[123] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2023-05-03 09:11:21 2023/05/03 12:11:21 wazuh-db[123] main.c:119 at main(): DEBUG: Wazuh home directory: /var/ossec
2023-05-03 09:11:24 2023/05/03 12:11:24 wazuh-remoted[387] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2023-05-03 09:11:24 2023/05/03 12:11:24 wazuh-remoted[387] main.c:130 at main(): DEBUG: Wazuh home directory: /var/ossec
2023-05-03 09:11:24 2023/05/03 12:11:24 wazuh-remoted[387] main.c:148 at main(): DEBUG: This is not a worker
2023-05-03 09:11:19 Starting Wazuh v4.4.2...
2023-05-03 09:11:21 Started wazuh-apid...
2023-05-03 09:11:21 Started wazuh-csyslogd...
2023-05-03 09:11:21 Started wazuh-dbd...
2023-05-03 09:11:21 Started wazuh-integratord...
2023-05-03 09:11:21 Started wazuh-agentlessd...
2023-05-03 09:11:21 Started wazuh-authd...
2023-05-03 09:11:22 Started wazuh-db...
2023-05-03 09:11:23 Started wazuh-execd...
2023-05-03 09:11:23 Started wazuh-analysisd...
2023-05-03 09:11:24 Started wazuh-syscheckd...
2023-05-03 09:11:25 Started wazuh-remoted...
2023-05-03 09:11:25 Started wazuh-logcollector...
2023-05-03 09:11:25 Started wazuh-monitord...
2023-05-03 09:11:26 Started wazuh-modulesd...
2023-05-03 09:11:27 Started wazuh-clusterd...
2023-05-03 09:11:29 Completed.
```
3. Copy Azure credential files into the master container using **docker cp** command
4. configure ossec.conf
```
  <wodle name="azure-logs">
    <disabled>no</disabled>
    <interval>10m</interval>
    <run_on_start>yes</run_on_start>

    <graph>
        <auth_path>/var/ossec/wodles/azure/azure.credentials</auth_path>
        <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>
        <request>
            <query>auditLogs/directoryaudits</query>
            <time_offset>30d</time_offset>
        </request>

        <request>
            <query>auditLogs/signIns</query>
            <time_offset>30d</time_offset>
        </request>

        <request>
            <query>auditLogs/provisioning</query>
            <time_offset>30d</time_offset>
        </request>
</graph>

    <log_analytics>
        <auth_path>/var/ossec/wodles/azure/credentials-analytics</auth_path>
        <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>

        <request>
            <workspace>d6b...efa</workspace>
            <query>AzureActivity</query>
            <time_offset>30d</time_offset>
        </request>
    </log_analytics>

    <storage>
        <auth_path>/var/ossec/wodles/azure/credentials-storage</auth_path>
        <tag>azure-activity</tag>

        <container name="frameworktestcontainer">
            <time_offset>30d</time_offset>
        </container>

        <container name="frameworktestcontainer"></container>

        <container name="frameworktestcontainer"/>
    </storage>
  </wodle>
```
### Test Execution
Restart the master node. 
Test result: ossec.log shows
```
2023/05/03 12:46:32 wazuh-modulesd:azure-logs: INFO: Module started.
2023/05/03 12:46:32 wazuh-modulesd:azure-logs: INFO: Starting fetching of logs.
2023/05/03 12:46:32 wazuh-modulesd:azure-logs: INFO: Starting Graphs log collection for the domain 'wazuh.onmicrosoft.com'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: ERROR: request_214950976: Returned error code: '1'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Finished Graphs log collection for request 'request_214950976'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: ERROR: request_776497670: Returned error code: '1'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Finished Graphs log collection for request 'request_776497670'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: ERROR: request_1605456294: Returned error code: '1'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Finished Graphs log collection for request 'request_1605456294'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Finished Graphs log collection for the domain 'wazuh.onmicrosoft.com'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Starting Log Analytics collection for the domain 'wazuh.onmicrosoft.com'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: ERROR: request_1691334341: Returned error code: '1'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Finished Log Analytics collection for request 'request_1691334341'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Finished Log Analytics collection for the domain 'wazuh.onmicrosoft.com'.
2023/05/03 12:46:33 wazuh-modulesd:azure-logs: INFO: Starting Storage log collection for 'azure-activity'.
2023/05/03 12:46:34 wazuh-modulesd:azure-logs: ERROR: frameworktestcontainer: Returned error code: '1'.
2023/05/03 12:46:34 wazuh-modulesd:azure-logs: INFO: Finished Storage log collection for container 'frameworktestcontainer'.
2023/05/03 12:46:34 wazuh-modulesd:azure-logs: ERROR: frameworktestcontainer: Returned error code: '1'.
2023/05/03 12:46:34 wazuh-modulesd:azure-logs: INFO: Finished Storage log collection for container 'frameworktestcontainer'.
2023/05/03 12:46:34 wazuh-modulesd:azure-logs: ERROR: frameworktestcontainer: Returned error code: '1'.
2023/05/03 12:46:34 wazuh-modulesd:azure-logs: INFO: Finished Storage log collection for container 'frameworktestcontainer'.
2023/05/03 12:46:34 wazuh-modulesd:azure-logs: INFO: Finished Storage log collection for 'azure-activity'.
```
7.

**Comment by EduLeon12** [comment]
# Docker Listener Test results


The [logall_json option](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall_json) must be enabled in the `ossec.conf` file before starting the tests.

## Test docker listener using ossec.conf in a manager

Apply the following configuration, restart Wazuh and check ossec.log to verify the module is running
```
<wodle name="docker-listener">
    <interval>30s</interval>
    <attempts>5</attempts>
    <run_on_start>yes</run_on_start>
    <disabled>no</disabled>
</wodle>
```

<details><summary> Config output</summary>

```
2023/05/03 18:56:39 wazuh-modulesd[275625] main.c:95 at main(): DEBUG: Created new thread for the 'docker-listener' module.
2023/05/03 18:56:39 wazuh-modulesd:docker-listener[275625] wm_docker.c:46 at wm_docker_main(): INFO: Module docker-listener started.
2023/05/03 18:56:39 wazuh-modulesd:docker-listener[275625] wm_docker.c:59 at wm_docker_main(): INFO: Starting to listening Docker events.
2023/05/03 18:56:39 wazuh-modulesd:docker-listener[275625] wm_docker.c:63 at wm_docker_main(): DEBUG: Launching command 'wodles/docker/DockerListener'
```

</details>

Run a container with no cache available
```
docker run ubuntu
```

<details><summary>Archives.json output</summary>

```
{"timestamp":"2023-05-03T20:17:42.481+0000","rule":{"level":3,"description":"Docker: Container charming_lovelace created","id":"87901","firedtimes":6,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"000","name":"wazuh-master"},"manager":{"name":"wazuh-master"},"id":"1683145062.2809165","cluster":{"name":"wazuh","node":"master-node"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"create\", \"id\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"create\", \"Actor\": {\"ID\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"charming_lovelace\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1683145062, \"timeNano\": 1683145062481265265}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"create","id":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","from":"ubuntu","Type":"container","Action":"create","Actor":{"ID":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","Attributes":{"image":"ubuntu","name":"charming_lovelace","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1683145062","timeNano":"1683145062481265152.000000"}},"location":"Wazuh-Docker"}
{"timestamp":"2023-05-03T20:17:42.482+0000","rule":{"level":3,"description":"Docker: Attached local standard input, output, and error streams to container charming_lovelace","id":"87922","firedtimes":6,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"wazuh-master"},"manager":{"name":"wazuh-master"},"id":"1683145062.2810475","cluster":{"name":"wazuh","node":"master-node"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"attach\", \"id\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"attach\", \"Actor\": {\"ID\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"charming_lovelace\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1683145062, \"timeNano\": 1683145062482121149}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"attach","id":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","from":"ubuntu","Type":"container","Action":"attach","Actor":{"ID":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","Attributes":{"image":"ubuntu","name":"charming_lovelace","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1683145062","timeNano":"1683145062482121216.000000"}},"location":"Wazuh-Docker"}
{"timestamp":"2023-05-03T20:17:42.742+0000","rule":{"level":3,"description":"Docker: Container charming_lovelace started","id":"87903","firedtimes":7,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"wazuh-master"},"manager":{"name":"wazuh-master"},"id":"1683145062.2812681","cluster":{"name":"wazuh","node":"master-node"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"start\", \"id\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"start\", \"Actor\": {\"ID\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"charming_lovelace\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1683145062, \"timeNano\": 1683145062742497033}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"start","id":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","from":"ubuntu","Type":"container","Action":"start","Actor":{"ID":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","Attributes":{"image":"ubuntu","name":"charming_lovelace","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1683145062","timeNano":"1683145062742497024.000000"}},"location":"Wazuh-Docker"}
{"timestamp":"2023-05-03T20:17:42.897+0000","rule":{"level":7,"description":"Docker: Container charming_lovelace received the action: die","id":"87924","firedtimes":7,"mail":false,"groups":["docker"],"gdpr":["IV_32.2"]},"agent":{"id":"000","name":"wazuh-master"},"manager":{"name":"wazuh-master"},"id":"1683145062.2814840","cluster":{"name":"wazuh","node":"master-node"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"die\", \"id\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"die\", \"Actor\": {\"ID\": \"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e\", \"Attributes\": {\"exitCode\": \"0\", \"image\": \"ubuntu\", \"name\": \"charming_lovelace\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1683145062, \"timeNano\": 1683145062897363328}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"die","id":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","from":"ubuntu","Type":"container","Action":"die","Actor":{"ID":"47a37ea95c8df6936a7ff05a0d215dab8c812f46446607aecdf227aae270415e","Attributes":{"exitCode":"0","image":"ubuntu","name":"charming_lovelace","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1683145062","timeNano":"1683145062897363456.000000"}},"location":"Wazuh-Docker"}

```
</details>


## Conclusions

The module worked as expected with the addition of the ASL configuration

**Comment by EduLeon12** [comment]
# Shuffle Test Results

The test assumes there is a functional Shuffle instance with an available webhook.

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected output was present in the `{wazuh_path}/logs/integrations.log` and `{wazuh_path}/logs/ossec.log` file.


## Test using ossec.conf in a manager

The following configuration has to be added to `ossec.conf`

```
<!-- Integration with Shuffle -->
<integration>
  <name>shuffle</name>
  <hook_url>shuffle_hook_url</hook_url> 
  <level>3</level>
  <alert_format>json</alert_format>
</integration>

```
Also, the configuration `integrator.debug=2` to `local_internal_options.conf `.

<details><summary>ossec.log</summary>
    
```
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: 
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Wed May 03 20:49:32 UTC 2023: # Sending message
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: 
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Wed May 03 20:49:32 UTC 2023: # In send msg
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: 
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Wed May 03 20:49:32 UTC 2023: # After send msg: <Response [200]>
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: 
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:425 at OS_IntegratorD(): DEBUG: Command ran successfully.
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:135 at OS_IntegratorD(): DEBUG: jqueue_next()
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:142 at OS_IntegratorD(): DEBUG: sending new alert.
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:276 at OS_IntegratorD(): DEBUG: file /tmp/shuffle-1683146972-512199745.alert was written.
2023/05/03 20:49:32 wazuh-integratord[44525] integrator.c:403 at OS_IntegratorD(): DEBUG: Running: integrations /tmp/shuffle-1683146972-512199745.alert  
```

</details>

<details><summary>integrations.log</summary>

```
Wed May 03 20:53:39 UTC 2023: # Starting
Wed May 03 20:53:39 UTC 2023: # Webhook
Wed May 03 20:53:39 UTC 2023: https://shuffler.io/api/v1/hooks/webhook
Wed May 03 20:53:39 UTC 2023: # File location
Wed May 03 20:53:39 UTC 2023: /tmp/shuffle-1683147219--1752981711.alert
Wed May 03 20:53:39 UTC 2023: # Processing alert
Wed May 03 20:53:39 UTC 2023: {'timestamp': '2023-05-03T20:53:36.470+0000', 'rule': {'level': 3, 'description': 'Docker: Command launched in container dev_wazuh-master_1. Action: "exec_start: /bin/sh -c /scripts/healthcheck.sh"', 'id': '87907', 'firedtimes': 275, 'mail': False, 'groups': ['docker'], 'gdpr': ['IV_32.2']}, 'agent': {'id': '000', 'name': 'wazuh-master'}, 'manager': {'name': 'wazuh-master'}, 'id': '1683147216.7651614', 'cluster': {'name': 'wazuh', 'node': 'master-node'}, 'decoder': {'name': 'json'}, 'data': {'integration': 'docker', 'docker': {'status': 'exec_start: /bin/sh -c /scripts/healthcheck.sh', 'id': 'afaa0044bafbf61ae37f35e4d0dd1575100c5a3f3a78dcc8fa15166633cbcdf6', 'from': 'dev-wazuh-manager', 'Type': 'container', 'Action': 'exec_start: /bin/sh -c /scripts/healthcheck.sh', 'Actor': {'ID': 'afaa0044bafbf61ae37f35e4d0dd1575100c5a3f3a78dcc8fa15166633cbcdf6', 'Attributes': {'com': {'docker': {'compose': {'config-hash': 'e242b52b8b8a62a6e11371b957175127bc0ad0f85c48b99d7ba8900476f171d4', 'container-number': '1', 'oneoff': 'False', 'project': 'dev', 'service': 'wazuh-master', 'version': '1.29.2'}}}, 'execID': 'a9141dc164c4a3eca9e48aa14e7355092332d39e58da72e77248004c1d8face6', 'image': 'dev-wazuh-manager', 'name': 'dev_wazuh-master_1', 'org': {'opencontainers': {'image': {'ref': {'name': 'ubuntu'}, 'version': '20.04'}}}}}, 'scope': 'local', 'time': '1683147216', 'timeNano': '1683147216469330944.000000'}}, 'location': 'Wazuh-Docker'}
Wed May 03 20:53:39 UTC 2023: # Generating message
Wed May 03 20:53:39 UTC 2023: {"severity": 1, "pretext": "WAZUH Alert", "title": "Docker: Command launched in container dev_wazuh-master_1. Action: \"exec_start: /bin/sh -c /scripts/healthcheck.sh\"", "text": null, "rule_id": "87907", "timestamp": "2023-05-03T20:53:36.470+0000", "id": "1683147216.7651614", "all_fields": {"timestamp": "2023-05-03T20:53:36.470+0000", "rule": {"level": 3, "description": "Docker: Command launched in container dev_wazuh-master_1. Action: \"exec_start: /bin/sh -c /scripts/healthcheck.sh\"", "id": "87907", "firedtimes": 275, "mail": false, "groups": ["docker"], "gdpr": ["IV_32.2"]}, "agent": {"id": "000", "name": "wazuh-master"}, "manager": {"name": "wazuh-master"}, "id": "1683147216.7651614", "cluster": {"name": "wazuh", "node": "master-node"}, "decoder": {"name": "json"}, "data": {"integration": "docker", "docker": {"status": "exec_start: /bin/sh -c /scripts/healthcheck.sh", "id": "afaa0044bafbf61ae37f35e4d0dd1575100c5a3f3a78dcc8fa15166633cbcdf6", "from": "dev-wazuh-manager", "Type": "container", "Action": "exec_start: /bin/sh -c /scripts/healthcheck.sh", "Actor": {"ID": "afaa0044bafbf61ae37f35e4d0dd1575100c5a3f3a78dcc8fa15166633cbcdf6", "Attributes": {"com": {"docker": {"compose": {"config-hash": "e242b52b8b8a62a6e11371b957175127bc0ad0f85c48b99d7ba8900476f171d4", "container-number": "1", "oneoff": "False", "project": "dev", "service": "wazuh-master", "version": "1.29.2"}}}, "execID": "a9141dc164c4a3eca9e48aa14e7355092332d39e58da72e77248004c1d8face6", "image": "dev-wazuh-manager", "name": "dev_wazuh-master_1", "org": {"opencontainers": {"image": {"ref": {"name": "ubuntu"}, "version": "20.04"}}}}}, "scope": "local", "time": "1683147216", "timeNano": "1683147216469330944.000000"}}, "location": "Wazuh-Docker"}}
Wed May 03 20:53:39 UTC 2023: # Sending message
Wed May 03 20:53:39 UTC 2023: # In send msg
Wed May 03 20:53:39 UTC 2023: # After send msg: <Response [200]>

```

</details>

<details>
   <summary> Shuffle UI </summary>
![Wazuh-alert](https://user-images.githubusercontent.com/60372646/236047791-e11ea870-5b10-409c-9575-6e5aafed2ba6.png)

</details>

### Conclusions
The integration works as expected, sending the alerts to Shuffle.
